daffodil-core/src/main/scala/org/apache/daffodil/grammar/ElementBaseGrammarMixin.scala

```diff
@@ -577,17 +577,23 @@ trait ElementBaseGrammarMixin
     lazy val allowedValue = allowedValueArg
     if (this.isOutputValueCalc)
       SimpleTypeRetry(this, allowedValue)
-    else if (this.isInstanceOf[PrefixLengthQuasiElementDecl] &&
-      this.asInstanceOf[PrefixLengthQuasiElementDecl].detachedReference.impliedRepresentation == Representation.Text)
-      // If an element has text representation and has a prefixed length, it
+    else if (this.isInstanceOf[PrefixLengthQuasiElementDecl]) {
+      //
+      // This predicate used to check the detachedReference for representation text.
+      // Seems like a mistake. It's just about variable length here.
+      // You can have prefixed length on hexBinary.
+      //
+      // && this.asInstanceOf[PrefixLengthQuasiElementDecl].detachedReference.impliedRepresentation == Representation.Text)
+      //
```

Comment:
Agreed, I think this is wrong, I'm not sure keeping this comment around adds anything. I think it's just confusing.

Reply:
I'll take out this comment.